internals/cli/ui/ask.go

```diff
@@ -51,7 +51,7 @@ func AskWithDefault(io IO, question, defaultValue string) (string, error) {
 
 // AskSecret prints out the question and reads back the input,
 // without echoing it back. Useful for passwords and other sensitive inputs.
-func AskSecret(io IO, question string) (string, error) {
+func AskSecret(io IO, passwordReader PasswordReader, question string) (string, error) {
```

Comment:
Do we want the password reader to be configurable? We could also call `NewPasswordReader` internally, keeping usage simple (and backwards compatible).

Reply:
See also the amount of files you have to touch to introduce this change.

Reply:
I passed the `PasswordReader` to allow injecting it for testing purposes. Without this, one of the `Write` command tests will no longer be possible after this rewrite.

Reply:
You mean the `"ask secret success"` and "ask secret error" testcase, right? 

Let's first think about what it is we're trying to test here. It's a unit test for the write command, so we only want to test the logic in the write command, not test the internals of the `AskSecret` function.

If we make the password reader internal to the `AskSecret` function, the test that verifies `AskSecret` handles the output of the password reader correctly should be a test for `AskSecret`. The unit tests for the write command can focus on the interaction between the write command and the `AskSecret` function. Can we mock `AskSecret` instead?

Reply:
Good point! I'll mock `AskSecret` then!